### ISSUE TITLE
Inline Temporary fails to simplify `global::` in top-level statements

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#44420]
**Steps to Reproduce**:

Inline `val2`:

```csharp
int val = 0;
int val2 = val + 1;
System.Console.WriteLine(val2);
```

**Expected Behavior**:

```csharp
int val = 0;
System.Console.WriteLine(val + 1);
```

**Actual Behavior**:

```csharp
int val = 0;
global::System.Console.WriteLine(val + 1);
```

**Additional Notes**:

When the reducer compares the bound symbols for `global::System.Console.WriteLine` (original semantic model) and `System.Console.WriteLine` (speculative semantic model), the former is binding to `WriteLine(int)` but the latter is reporting an overload resolution failure with 19 candidates.
